server/schedule/operator/step.go

```diff
@@ -197,19 +208,19 @@ func (pl PromoteLearner) String() string {
 
 // IsFinish checks if current step is finished.
 func (pl PromoteLearner) IsFinish(region *core.RegionInfo) bool {
-	if p := region.GetStoreVoter(pl.ToStore); p != nil {
-		if p.GetId() != pl.PeerID {
-			log.Warn("obtain unexpected peer", zap.String("expect", pl.String()), zap.Uint64("obtain-voter", p.GetId()))
+	if peer := region.GetStoreVoter(pl.ToStore); peer != nil {
+		if peer.GetId() != pl.PeerID {
+			log.Warn("obtain unexpected peer", zap.String("expect", pl.String()), zap.Uint64("obtain-voter", peer.GetId()))
 		}
-		return p.GetId() == pl.PeerID
+		return peer.GetId() == pl.PeerID
 	}
 	return false
 }
 
 // CheckSafety checks if the step meets the safety properties.
 func (pl PromoteLearner) CheckSafety(region *core.RegionInfo) error {
 	peer := region.GetStorePeer(pl.ToStore)
-	if peer == nil {
+	if peer.GetId() != pl.PeerID {
```

Comment:
Do we still need to check if peer is `nil` here?

Reply:
The `GetXX` function of `ProtoBuf` has its own `nil` judgment.